osquery/events/linux/inotify.cpp

```diff
@@ -241,10 +241,10 @@ bool INotifyEventPublisher::addMonitor(const std::string& path,
   if (recursive && isDirectory(path).ok()) {
     std::vector<std::string> children;
     // Get a list of children of this directory (requested recursive watches).
-    listDirectoriesInDirectory(path, children);
+    listDirectoriesInDirectoryRecursively(path, children);
 
     for (const auto& child : children) {
-      addMonitor(child, recursive);
+      addMonitor(child, false);
```

Comment:
I think this is the error, swapping this to `false` and appending "**" to the substring at [line 67](https://github.com/lexelby/osquery/blob/inotify-recursion/osquery/events/linux/inotify.cpp#L67) should scope the recursion to ONLY the filesystem API.

I'm not a huge fan of adding the additional `listDirectoriesInDirectoryRecursively` method, unless we have some tests that "name/**" is functionally equivalent to "name**".